### PR TITLE
Handle validation errors raised by `autocomplete_create` method

### PR DESCRIPTION
**Pull request by chigby**
Right now, zero error handling is done when we invoke this method. There are probably a lot of ways we could go about this, but building in handling of validation error (and also returning the results of that error, if there are any), allows users of this feature to use fairly standard Django practices to validate their object before saving and have that be understood by the autocomplete view. One example of how to do this is given in the updated `testapp` code.

Fixes https://github.com/wagtail/wagtail-autocomplete/issues/134

### CODE REVIEW

**Review by SaptakS — CHANGES_REQUESTED** [review]
The code for raising the backend error looks good to me and makes sense. I do however think that there should be some error handling on frontend side so the user sees some error message. I know this increases the scope of the ticket a lot but it's not only a confusing experience for the users of the admin, but is also an accessibility requirement https://www.w3.org/WAI/WCAG21/Understanding/error-identification.html

We can probably add some reasoning based on the status [here](https://github.com/wagtail/wagtail-autocomplete/blob/main/client/src/utils/client.js#L19-L27) and then pass the error from [here](https://github.com/wagtail/wagtail-autocomplete/blob/main/client/src/utils/client.js#L72-L73), and then handle the error again using `.catch` [here](https://github.com/wagtail/wagtail-autocomplete/blob/main/client/src/components/AutocompleteInput/AutocompleteInput.js#L140-L157). 

I think given AutocompleteInput is an input field component, we should definitely have a way to show errors in the component (which is missing right now). According to me, just an error message after the input field should be okay, but @harrislapiroff might have better design ideas.

**Comment by chigby** [comment]
Added a small error message display on the front-end. Not sure if it's really the best, but, well, it's something. Suggestions welcome. 
image

File: client/src/components/AutocompleteInput/errors.scss [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+  color: $red;
+}
+
+
+// .help-block {
+//  clear: both;
+//  color: var(--w-color-grey-600);
+//  margin: 1em 0;
+//  padding: 1em;
+// }
```
Comment (SaptakS) [inline client/src/components/AutocompleteInput/errors.scss:11]: @chigby I think we can remove this comments?

Comment (chigby) [reply client/src/components/AutocompleteInput/errors.scss:11]: oops

**Review by SaptakS — APPROVED** [review]
Looks good to me.

**Pull request merged**